commitlintplugins.test.ts

```diff
@@ -4,12 +4,87 @@ function runCommitLintOnMsg(inputMsg: string) {
     return spawnSync('npx', ['commitlint', '--verbose'], { input: inputMsg });
 }
 
+test('body-prose1', () => {
+    let commitMsgWithLowercaseBodyStart =
+        "foo: this is only a title" + "\n\n" + "bla blah bla.";
+    let bodyProse1 = runCommitLintOnMsg(commitMsgWithLowercaseBodyStart);
+    expect(bodyProse1.status).not.toBe(0);
+});
+
+
+test('body-prose2', () => {
+    let commitMsgWithNumbercaseBodyStart =
+        "foo: this is only a title" + "\n\n" + "1234 bla blah bla.";
+    let bodyProse2 = runCommitLintOnMsg(commitMsgWithNumbercaseBodyStart);
+    expect(bodyProse2.status).toBe(0);
+});
+
+
+test('body-prose3', () => {
+    let commitMsgWithUrl =
+        "foo: this is only a title" + "\n\n" + "someUrl://blahblah.com";
+    let bodyProse3 = runCommitLintOnMsg(commitMsgWithUrl);
+
+    // because URLs can bypass the limit
+    expect(bodyProse3.status).toBe(0);
+});
+
+
+test('body-prose4', () => {
+    let commitMsgWithFootnoteUrl =
+        "foo: this is only a title" + "\n\n" + "Bla blah[1] bla.\n\n[1] someUrl://blahblah.com";
+    let bodyProse4 = runCommitLintOnMsg(commitMsgWithFootnoteUrl);
+
+    // because URLs in footer can bypass the limit
+    expect(bodyProse4.status).toBe(0);
+});
+
+
+test('body-prose5', () => {
+    let commitMsgWithBugUrl =
+        "foo: this is only a title" + "\n\n" + "Fixes someUrl://blahblah.com";
+    let bodyProse5 = runCommitLintOnMsg(commitMsgWithBugUrl);
+
+    // because URLs in "Fixes <URL>" sentence can bypass the limit
+    expect(bodyProse5.status).toBe(0);
+});
+
+
+test('body-prose6', () => {
+    let commitMsgWithBlock =
+        "foo: this is only a title" + "\n\n" + "Bar baz.\n```\n" + 'if (foo) {bar(); }' + "\n```";
+    let bodyProse6 = runCommitLintOnMsg(commitMsgWithBlock);
+
+    // because ```blocks surrounded like this``` can bypass the rule
+    expect(bodyProse6.status).toBe(0);
+});
+
+
+test('body-prose7', () => {
+    let commitMsgWithParagraphEndingWithColon =
+        "foo: this is only a title" + "\n\n" + "Bar baz:\n\nBlah blah.";
+    let bodyProse7 = runCommitLintOnMsg(commitMsgWithParagraphEndingWithColon);
+
+    // because paragraphs can end with a colon
+    expect(bodyProse7.status).toBe(0);
+});
+
+
+test('body-prose8', () => {
+    let commitMsgWithBlockParagraph=
+        "foo: this is only a title" + "\n\n" + "Bar baz:\n\n```\ntype Foo =\n    string\n```";
+    let bodyProse8 = runCommitLintOnMsg(commitMsgWithBlockParagraph);
+
+    // because paragraphs that only consist of a block can bypass the rule
```

Comment:
let's just say that a block is not a paragraph

Reply:
actually, shouldnt this be covered by body-prose6 test? I see this test as redundant

Reply:
> let's just say that a block is not a paragraph

Do you mean this is not a valid body?

Reply:
I mean to change the comment

Reply:
Thing is, I think body-prose6 should have \n\n as separation between first paragraph and the block. If you change it this way, you understand that this test is the same right?

Reply:
Right. Done!